pyproject.toml

```diff
@@ -29,7 +29,6 @@ classifiers = [
 ]
 dependencies = [
     "certifi",
-    "httpcore>=0.17.2,<0.18.0",
```

Comment:
In my understanding, this is runtime dependencies, remove `httpcore` from here will cause users running `pip intsall httpx` not installing `httpcore`

Reply:
So the user installs httpx but is unable to use it because it also requires the installation of httpcore? 
Is there any motivation to do so?

Reply:
> So the user installs httpx but is unable to use it because it also requires the installation of httpcore?

Yes, users will find they can't use `httpx` because `httpx` import `httpcore`, but `httpx` didn't specify `httpcore` as dependencies. 

`requirements.txt` are actually only for developing and testing.

> Is there any motivation to do so?

I didn't understand this question...

Reply:
> I mean, what problem does it solve?

`httpx` require `httpcore`, and remove this line also remove `httpcore` from install_requires

you can try this:
```shell
python -m venv .venv
pip install  https://github.com/karosis88/httpx/archive/refs/heads/change-response-extensions-type.zip # this is change-response-extensions-type branch of your git repo
python -c 'import httpx'
```

Reply:
Sorry, but I don't understand why we should do that. Why should we remove httpcore as a dependency and install it independently?

Reply:
It's not "we should do", it's "what you have done in this PR"

Reply:
Your PR (this deletion line I comment on) remove `httpcore` as a dependency.

Reply:
> In my understanding, this is runtime dependencies, remove `httpcore` from here will cause users running `pip intsall httpx` not installing `httpcore`

Sorry, I thought you were suggesting that httpcore be removed here :pray:

Reply:
> I'm saying "don't do it".

Of course, we won't do that; it was just a temporary commit that will be reverted in the future; we do it to test with the httpcore master.

Reply:
> I'm saying "don't do it".

Of course, we won't do that; it was just a temporary commit that will be reverted in the future; we do it to test with the httpcore master.

Reply:
> > I'm saying "don't do it".
> 
> Of course, we won't do that; it was just a temporary commit that will be reverted in the future; we do it to test with the httpcore master.

you can still add `httpcore @ git+https://github.com/encode/httpcore@master` to requirements.txt for testing, it will override the `pyproject.toml#dependencies` and install master branch of httpcore.